src/borg/testsuite/archiver.py

```diff
@@ -3731,9 +3731,88 @@ def do_asserts(output, can_compare_ids):
             if are_hardlinks_supported():
                 assert 'input/hardlink_target_replaced' not in output
 
+        def do_json_asserts(output, can_compare_ids):
+            def get_changes(filename, data):
+                chgsets = [j['changes'] for j in data if j['path'] == filename]
+                assert len(chgsets) < 2
+                # return a flattened list of changes for given filename
+                return [chg for chgset in chgsets for chg in chgset]
+
+            # convert output to list of json lines
```

Comment:
convert json lines to list of dicts

Reply:
dang your quick! fastest reviewer in the west!

Reply:
just pushed all your suggestions.